ddtrace/_trace/tracer.py

```diff
@@ -1125,18 +1126,44 @@ def _is_span_internal(span):
 
     def _on_global_config_update(self, cfg, items):
         # type: (Config, List) -> None
-        if "_trace_sample_rate" in items:
-            # Reset the user sampler if one exists
-            if cfg._get_source("_trace_sample_rate") != "remote_config" and self._user_sampler:
-                self._sampler = self._user_sampler
+
+        # sampling configs always come as a pair
+        if "_trace_sample_rate" in items and "_trace_sampling_rules" in items:
```

Comment:
Putting this complex logic in a `sampling.py` helper function might underline the fact that it's complex and not directly related to the tracer. It would also reduce the cognitive load incurred on editors of this file.

Reply:
I like that idea too, but it leads to a circular import because I need to import DatadogSampler into sampling.py. I guess I can just return the sampling_rules and sample_rate though.

Reply:
Ah, can't do that either, I will break it into a separate method though.